src/readBuffer.js

```diff
@@ -1,12 +1,19 @@
+const EOF = new Error('EOF')
```

Comment:
An error should be constructed at the time it is thrown, because the instatiation captures the stack trace.

Reply:
OK.